### PR TITLE
VB-3954 Fix to double booking issue.

### PR DESCRIPTION
**Pull request by AledWEvans**
Double booking issue

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/builder/VisitDtoBuilder.kt [hunk 1]
```diff
@@ -37,4 +42,17 @@ class VisitDtoBuilder {
...
+  private fun getApplicationReference(
+    visitEntity: Visit,
+  ): String {
+    val application = visitEntity.getLastCompletedApplication()
+    return application?.let {
+      it.reference
+    } ?: run {
+      // This catches an issues when two requests from the booking occur at the same time see bookVisit method in visit service
+      LOG.error("Visit ${visitEntity.reference} should have a completed application")
+      "toBeAnnounced"
```
Comment (adaviesMOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/builder/VisitDtoBuilder.kt:55]: Will this set the application reference = "toBeAnnounced" ? Is that what we want here?

**Review by adaviesMOJ — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/WebTestClientConfig.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  // This is to allow calls to web test client to happen at the same time rather than serially
+  // This is how things in real world!
+  override fun customize(builder: Builder) {
+    builder.filter { clientRequest: ClientRequest?, next: ExchangeFunction ->
+      next.exchange(
+        clientRequest,
+      ).subscribeOn(Schedulers.parallel())
+    }
+  }
+}
```
Comment (dhirajshettymoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/WebTestClientConfig.kt:21]: nice!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/DoubleBookingVisitTest.kt [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    roleVisitSchedulerHttpHeaders = setAuthorisation(roles = listOf("ROLE_VISIT_SCHEDULER"))
+
+    reservedApplication = applicationEntityHelper.create(sessionTemplate = sessionTemplateDefault, completed = false)
+    applicationEntityHelper.createContact(application = reservedApplication, name = "Jane Doe", phone = "01234 098765")
+    applicationEntityHelper.createVisitor(application = reservedApplication, nomisPersonId = 321L, visitContact = true)
+    reservedApplication = applicationEntityHelper.save(reservedApplication)
+  }
+
+  @Test
+  fun `Book multiple visits using same application reference results in the same booking returned`() {
```
Comment (dhirajshettymoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/DoubleBookingVisitTest.kt:36]: difficult to test this issue, good test!

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Review by adaviesMOJ — APPROVED** [review]

**Pull request merged**